File: base/src/test/java/io/quarkus/code/model/ProjectDefinitionTest.java
import io.quarkus.code.model.ProjectDefinition;
import org.junit.jupiter.api.Test;

import java.util.regex.Pattern;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class ProjectDefinitionTest {

    @Test
    public void testClassNamePattern() {
        assertTrue(Pattern.compile(ProjectDefinition.CLASSNAME_PATTERN).asPredicate().test("de.if.ExampleResource"));
    }
}

File: src/test/kotlin/io/quarkus/code/model/ProjectDefinitionTest.kt
package io.quarkus.code.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.regex.Pattern

internal class ProjectDefinitionTest {

    @Test
    internal fun testClassNamePattern() {
        assertThat(Pattern.compile(ProjectDefinition.CLASSNAME_PATTERN).asPredicate()).accepts("de.if.ExampleResource")
    }
}